src/Microsoft.AspNet.Authentication.OpenIdConnect/OpenIdConnectAuthenticationHandler.cs

```diff
@@ -321,6 +333,45 @@ protected override AuthenticationTicket AuthenticateCore()
                     _configuration = await Options.ConfigurationManager.GetConfigurationAsync(Context.RequestAborted);
                 }
 
+                if (!string.IsNullOrWhiteSpace(message.Code))
```

Comment:
I moved the notification so that user can plug in their own code for exchanging "code" for tokens in case of "Code only flow". Now, I think they can do so in messageReceivedNotification of by overriding RedeemAuthorizationCode method (which requires more work!). I moved the authorizationCodeReceivedNotification back to where it was earlier.